packages/components/src/utils/validationSchemes.test.ts

```diff
@@ -0,0 +1,118 @@
+import { type Network } from "@umami/tezos";
+
+import { getNetworkValidationScheme } from "./validationSchemes";
+
+describe("getNetworkValidationScheme", () => {
+  const mockAvailableNetworks: Network[] = [
+    {
+      name: "Existing Network",
+      rpcUrl: "https://example.com",
+      tzktApiUrl: "https://api.example.com",
+      tzktExplorerUrl: "https://explorer.example.com",
+    },
+  ];
+
+  it("should validate a correct network input", () => {
+    const schema = getNetworkValidationScheme(mockAvailableNetworks);
+    const validInput = {
+      name: "New Network",
+      rpcUrl: "https://rpc.example.com",
+      tzktApiUrl: "https://api.tzkt.example.com",
+      tzktExplorerUrl: "https://explorer.tzkt.example.com",
+      buyTezUrl: "https://buy.example.com",
+    };
+
+    const result = schema.safeParse(validInput);
+    expect(result.success).toBe(true);
+  });
+
+  it("should reject a network with an existing name", () => {
+    const schema = getNetworkValidationScheme(mockAvailableNetworks);
+    const invalidInput = {
+      name: "Existing Network",
+      rpcUrl: "https://rpc.example.com",
+      tzktApiUrl: "https://api.tzkt.example.com",
+      tzktExplorerUrl: "https://explorer.tzkt.example.com",
+    };
+
+    const result = schema.safeParse(invalidInput);
+    expect(result.success).toBe(false);
+    if (!result.success) {
+      expect(result.error.issues[0].message).toBe("Network with this name already exists");
+    }
+  });
+
+  it("should reject invalid URLs", () => {
+    const schema = getNetworkValidationScheme(mockAvailableNetworks);
+    const invalidInput = {
+      name: "Invalid URL Network",
+      rpcUrl: "http://insecure.com",
+      tzktApiUrl: "not-a-url",
+      tzktExplorerUrl: "ftp://invalid-protocol.com",
+    };
+
+    const result = schema.safeParse(invalidInput);
+    expect(result.success).toBe(false);
+    if (!result.success) {
```

Comment:
just curious how result.success can be not false here? 🤔

Reply:
Be the TypeScript overlord, use the `!` operator